From f096002d07cdf0d91fb06d8bb4135c0a0f39a3a7 Mon Sep 17 00:00:00 2001
From: Robert Mustacchi <rm@joyent.com>
Date: Fri, 26 Aug 2016 22:22:41 +0000
Subject: [PATCH] OS-5635 hostname(1) manual page should talk about persistence
 Reviewed by: Trent Mick <trent.mick@joyent.com> Reviewed by: Cody Mello
 <melloc@joyent.com>

---
 usr/src/man/man1/hostname.1 | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/usr/src/man/man1/hostname.1 b/usr/src/man/man1/hostname.1
index 5b63dab64a..75a24a0437 100644
--- a/usr/src/man/man1/hostname.1
+++ b/usr/src/man/man1/hostname.1
@@ -1,8 +1,9 @@
 .\" Copyright (c) 1992, Sun Microsystems, Inc.
+.\" Copyright 2016 Joyent, Inc.
 .\" The contents of this file are subject to the terms of the Common Development and Distribution License (the "License").  You may not use this file except in compliance with the License.
 .\" You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE or http://www.opensolaris.org/os/licensing.  See the License for the specific language governing permissions and limitations under the License.
 .\" When distributing Covered Code, include this CDDL HEADER in each file and include the License file at usr/src/OPENSOLARIS.LICENSE.  If applicable, add the following below this CDDL HEADER, with the fields enclosed by brackets "[]" replaced with your own identifying information: Portions Copyright [yyyy] [name of copyright owner]
-.Dd May 12, 2016
+.Dd Aug 26, 2016
 .Dt HOSTNAME 1
 .Os
 .Sh NAME
@@ -19,6 +20,17 @@ command prints the name of the current host, as given before the
 .Xr login 1
 prompt. The super-user can set the hostname by giving
 .Ar name-of-host .
+.Pp
+While setting the hostname changes the hostname for the running system,
+it does not persist the change. The proper way to update the hostname
+will vary depending on how the system is configured. For most systems,
+the persistent hostname is stored in the file
+.Pa /etc/nodename .
+See
+.Xr nodename 4
+for more information on how to update it. Note, the
+.Pa /etc/nodename
+file may be ignored in cases where the system is configured to use DHCP.
 .Sh OPTIONS
 .Bl -tag -width Ds
 .It Fl s
@@ -26,4 +38,5 @@ Prints only the part of the hostname preceding the first dot.
 .El
 .Sh SEE ALSO
 .Xr uname 1 ,
+.Xr nodename 4 ,
 .Xr attributes 5
-- 
2.21.0

